src/OpenCL.cpp

```diff
@@ -772,19 +818,29 @@ void OpenCL::initialize(void) {
     //std::string sourceCode(std::istreambuf_iterator<char>(sourceFile),
     //                       (std::istreambuf_iterator<char>()));
 
+
     // Make program of the source code in the context
     try {
         m_program = cl::Program(sourceCode_config
-                                + sourceCode_convolve1
                                 + sourceCode_convolve3
-                                + sourceCode_utility);
+                                + sourceCode_utility
+                                + sourceCode_sgemm);
     } catch (const cl::Error &e) {
         myprintf("Error getting kernels: %s: %d", e.what(), e.err());
         throw;
     }
+
+    m_cl_args = cl_args;
+
+    auto t = Tuner();
+    auto sgemm_tuners = t.load_sgemm_tuners(128, 100, 128, 16);
```

Comment:
Magic constants? Also you're tuning 128x6 here instead of the actual network size?